tools/sort.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+# Shell script equivalent of Metamath SORT tool
+# Mario Carneiro 7-Jan-2022
+
+usage() {
+  cat << HELP
+SORT - Sort the lines in a file with key starting at specified string
+
+This command sorts a file, comparing lines starting at a key string.
+If the key string is blank, the line is compared starting at column 1.
+If a line doesn't contain the key, it is compared starting at column 1.
+Syntax:  sort.sh KEY < INFILE > OUTFILE
+HELP
+}
+
+if [ $# -eq 1 ] && [ "$1" = "-h" -o "$1" = "--help" ]; then usage; exit; fi
+if [ $# -ne 1 ]; then usage; exit 1; fi
+
+if [ "$1" == "" ]; then
+  sort
+else
+  awk -v matchKey="$1" '{
+    n = index($0, matchKey);
+    key = n ? substr($0, n) : $0;
+    print key "\1" $0;
+  }' | sort -k1,1 -t $'\x01' | sed 's/.*\x01//'
+fi
```

Comment:
Nice.
A slightly better choice of the delimiter is 0x1F, see here https://stackoverflow.com/questions/492090/least-used-delimiter-character-in-normal-text-ascii-128.  This could be the default for an optional second parameter of sort.sh